### PR TITLE
Enhance find command with block field (and update DG)

### LINKED ISSUE #118: As a Hall admin finding residents, I want to search for students in a block, so that I can quickly find out which students are in which blocks.

### LINKED ISSUE #124: Update DG with enhanced find command

### PR DESCRIPTION
**Pull request by athuyaoo**
image
image

Users can now filter by block

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -345,6 +345,52 @@ Given below is a step-by-step usage scenario and how the listing all student gro
...
+
+Pros | Cons
+-----| -----
+\+ Memory is freed earlier as list of predicates is converted immediately | - Makes testing of equal find commands more difficult
+
+* **Alternative 2**: Create the predicate during `FindCommand#execute()`
+
+Pros | Cons
+-----|-----
+\+ It is easier to compare equality for `FindCommand` objects<br/>/+ It is easier to test | - More memory is needed to store the list of predicates for longer period of time
```
Comment (schoolex) [inline docs/DeveloperGuide.md]: Is there a formatting issue with the escape character

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -345,6 +345,52 @@ Given below is a step-by-step usage scenario and how the listing all student gro
...
+2. The `UI` handles the input and calls `LogicManager#execute()` to execute it.
+3. The `AddressBookParser#parseCommand()` is called to parse the input, which calls `FindCommandParser#parseCommand()`,
+subsequently returning a `FindCommand` with the associated list of predicates.
+4. The `FindCommand` calls `FindCommand#execute()` which forms a predicate that fulfill every predicate from the list.
+5. The predicate updates the filtered list of all residents by calling `Model#updateFilteredPersonList()`.
+The resulting filtered list matches this predicate, which means that it matches all predicates in the original list.
+7. The `UI` displays the result in the result box.
+
+The following sequence diagram shows how finding students works:
+![Find Sequence Diagram](images/FindSequenceDiagram.png)
```
Comment (schoolex) [inline docs/DeveloperGuide.md:370]: The argument in the method does not match step 1. Can consider shortening it to `find`

**Review by schoolex — COMMENTED** [review]
Almost there, some nits to fix

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (athuyaoo) [reply docs/DeveloperGuide.md:370]: Thanks, resolved.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (athuyaoo) [reply docs/DeveloperGuide.md]: Thanks, resolved.

**Review by schoolex — APPROVED** [review]
lgtm

**Pull request merged**